src/client/search-init.ts

```diff
@@ -10,22 +10,20 @@ input.addEventListener("focus", load, {once: true});
 input.addEventListener("keydown", load, {once: true});
 
 // Focus on meta-K and /
-const toggle = document.querySelector("#observablehq-sidebar-toggle")!;
+const sidebar = document.querySelector("#observablehq-sidebar")!;
 addEventListener("keydown", (event) => {
   if (
     (event.code === "KeyK" && event.metaKey && !event.altKey && !event.ctrlKey) ||
     (event.key === "/" && !event.metaKey && !event.altKey && !event.ctrlKey && event.target === document.body)
   ) {
     // Force the sidebar to be temporarily open while the search input is
-    // focused. (We can’t use :focus-within because the sidebar isn’t focusable
-    // while it is invisible, and we don’t want to keep the sidebar open
-    // persistently after you blur the search input.)
-    toggle.classList.add("observablehq-sidebar-open");
+    // focused. The class makes the sidebar visible so the input can be focused,
+    // then the :focus-within selector takes over and keeps it open until the
+    // user blurs the input.
+    sidebar.classList.add("observablehq-sidebar-open");
```

Comment:
Could you do this by temporarily toggling the checkbox? Then you may not need the `observablehq-sidebar-open` class at all?

Reply:
yes!

Note: the code I pushed now assumes that we use 'visibility:hidden' to hide the side bar, and uses getComputedStyle to know what state we're in. We could skip this step and click on the toggle in any case (even when it's doing the opposite and starts closing the bar), because thanks to the _transition_ it is still visible. But I felt it fragile to rely on a css transition.